src/internet_identity/tests/tests.rs

```diff
@@ -53,11 +53,35 @@ mod upgrade_tests {
         assert_eq!(retrieved_device_data, vec![device_data_1()]);
     }
 
-    /// Test to verify that anchors number range cannot be changed on upgrade.
+    /// Test to verify that anchors number range can be changed on upgrade.
     #[test]
-    fn ii_upgrade_should_not_allow_change_of_user_ranges() {
+    fn should_allow_change_of_user_range_on_upgrade() -> Result<(), CallError> {
         let env = env();
-        let canister_id = install_ii_canister(&env, II_WASM_PREVIOUS.clone());
+        let canister_id = install_ii_canister(&env, II_WASM.clone());
+
+        upgrade_ii_canister_with_arg(
+            &env,
+            canister_id,
+            II_WASM.clone(),
+            Some(InternetIdentityInit {
+                assigned_user_number_range: Some((2000, 4000)),
+                archive_config: None,
+                canister_creation_cycles_cost: None,
+            }),
+        )?;
+
+        let stats = api::stats(&env, canister_id)?;
+        assert_eq!(stats.assigned_user_number_range, (2000, 4000));
+        Ok(())
+    }
+
+    /// Test to verify that the user range cannot be changed to modify anchor numbers of existing anchors.
+    #[test]
+    fn should_not_affect_existing_anchors_on_user_range_change() {
```

Comment:
ah, I misunderstood this at first. I assumed this was checking that anchors would still work after changing the range. Is there a test for that? (just asking)

Reply:
No, actually not. I will add one 👍